### ISSUE TITLE
Implement k-fold CV on graphs

### ISSUE THREAD

**Issue by lyashevska** [point-cloud-radar/bird-cloud-gnn#90]
Graphs are randomly divided into k equal-sized folds. The model is trained k times, each time using a different combination of k-1 folds as the training set. The model is evaluated on the remaining fold as the validation set, calculating the performance metrics of interest (e.g., accuracy). The performance results from each fold are averaged to obtain an overall assessment of the model's performance.
